src/plugins/data/common/index_patterns/index_patterns/index_patterns.ts

```diff
@@ -370,19 +354,14 @@ export class IndexPatternsService {
     const spec = this.savedObjectToSpec(savedObject);
     const { title, type, typeMeta } = spec;
 
-    const isFieldRefreshRequired = this.isFieldRefreshRequired(spec.fields);
-    let isSaveRequired = isFieldRefreshRequired;
     try {
-      spec.fields = isFieldRefreshRequired
-        ? await this.refreshFieldSpecMap(spec.fields || {}, id, spec.title as string, {
-            pattern: title as string,
-            metaFields: await this.config.get(UI_SETTINGS.META_FIELDS),
-            type,
-            rollupIndex: typeMeta?.params?.rollupIndex,
-          })
-        : spec.fields;
+      spec.fields = await this.refreshFieldSpecMap(spec.fields || {}, id, spec.title as string, {
```

Comment:
refreshFieldSpecMap => loadFieldSpecMap

failure to load the field list should fail to load the index pattern.